plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/RefsTable.java

```diff
@@ -0,0 +1,110 @@
+import io.trino.spi.connector.ColumnMetadata;
+import io.trino.spi.connector.ConnectorPageSource;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.FixedPageSource;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.SystemTable;
+import io.trino.spi.predicate.TupleDomain;
+import org.apache.iceberg.Table;
+
+import java.util.List;
+
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class RefsTable
+        implements SystemTable
+{
+    private final ConnectorTableMetadata tableMetadata;
+    private final Table icebergTable;
+
+    public RefsTable(SchemaTableName tableName, Table icebergTable)
+    {
+        this.icebergTable = requireNonNull(icebergTable, "icebergTable is null");
+
+        this.tableMetadata = new ConnectorTableMetadata(requireNonNull(tableName, "tableName is null"),
+                ImmutableList.<ColumnMetadata>builder()
+                        .add(new ColumnMetadata("name", VARCHAR))
+                        .add(new ColumnMetadata("type", VARCHAR))
+                        .add(new ColumnMetadata("snapshot_id", BIGINT))
+                        .add(new ColumnMetadata("max_reference_age_in_ms", BIGINT))
+                        .add(new ColumnMetadata("min_snapshots_to_keep", INTEGER))
+                        .add(new ColumnMetadata("max_snapshot_age_in_ms", BIGINT))
+                        .build());
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        return new FixedPageSource(buildPages(tableMetadata, icebergTable));
+    }
+
+    private static List<Page> buildPages(ConnectorTableMetadata tableMetadata, Table icebergTable)
+    {
+        PageListBuilder pagesBuilder = PageListBuilder.forTable(tableMetadata);
+
+        icebergTable.refs().forEach((refName, ref) -> {
+            pagesBuilder.beginRow();
+            pagesBuilder.appendVarchar(refName);
+            pagesBuilder.appendVarchar(ref.isBranch() ? "BRANCH" : "TAG");
+            pagesBuilder.appendBigint(ref.snapshotId());
+            if (ref.maxRefAgeMs() != null) {
+                pagesBuilder.appendBigint(ref.maxRefAgeMs());
+            }
+            else {
+                pagesBuilder.appendNull();
+            }
+
+            if (ref.minSnapshotsToKeep() != null) {
+                pagesBuilder.appendInteger(ref.minSnapshotsToKeep());
```

Comment:
You could add in `PageListBuilder` new methods:

- `appendInteger(Integer)`
- `appendBigint(Long)`

to encapsulate the `null` check and make the code in this class more concise.

Reply:
Added new methods as suggested